Adding `object.keys` built-in function.

kevinswiber opened the pull request:
The `object.keys` function will return all top-level keys on a given object.  Since object keys in Rego don't have the same restrictions as names in JSON name-value pairs, we also ensure support for non-string key types.

Closes #5363.

kevinswiber commented on topdown/object.go:
```diff
@@ -138,6 +138,17 @@ func builtinObjectGet(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Ter
+		return err
+	}
+
+	keys := ast.NewArray(object.Keys()...)
```
Should this return a Set instead of an Array?  The keys are going to be de-duped and ordered anyway.

srenatus replied:
[nit] We could use ArrayTerm instead
```
$ git grep -C2 'func.*ArrayTerm'
ast/term.go-
ast/term.go-// ArrayTerm creates a new Term with an Array value.
ast/term.go:func ArrayTerm(a ...*Term) *Term {
ast/term.go-    return NewTerm(NewArray(a...))
ast/term.go-}
```

srenatus reviewed (commented):
This is a very comprehensive first contribution 👏 I think it's mostly good to go, except for switching to a set result.

kevinswiber commented:
@srenatus Thanks for the feedback.  I changed the return type from an Array to a Set, and I cleaned up the tests.  I was wondering why they were so weird-looking.  Makes sense they were generated.  :joy:  Let me know if there's anything else you see!

srenatus reviewed (approved):
💯 nicely done! Thanks a lot.
